lib/reindex/s3_bucketd.py

```diff
@@ -504,10 +518,12 @@ def log_report(resource, name, obj_count, total_size):
 
     stale_buckets = set()
     recorded_buckets = set(get_resources_from_redis(redis_client, 'buckets'))
-    if not options.bucket:
+    if not options.bucket and not options.account:
         stale_buckets = recorded_buckets.difference(observed_buckets)
     elif options.bucket:
         stale_buckets = { b for b in options.bucket if b not in observed_buckets }
+    elif options.account:
+        _log.warning('Stale buckets will not be cleared when using the --account or --account-file flags')
```

Comment:
It might be slightly reorganized for simplicity as

```
if options.bucket:
    ...
elif options.account:
    ...
else:
    # neither bucket nor account
    ...
```